server/app/api/v1/agents.py

```
@@ -0,0 +1,29 @@
+    try:
+        agent_manager = get_agent_manager()
+        agents = agent_manager.get_all_agents()
+        return [
+            AgentResponse(agent_id=id, agent_name=name)
+            for id, name in agents
+        ]
+    except Exception as e:
+        raise HTTPException(
+            status_code=500,
+            detail=f"Error fetching agents: {str(e)}"
+        )
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)